src/reportportal/rp_api_client.py

```
@@ -64,6 +65,59 @@ def build_headers(self) -> Dict[str, str]:
+        if isinstance(timestamp, str):
+            try:
+                dt = datetime.fromisoformat(timestamp.replace('Z', '+00:00'))
+                return int(dt.timestamp() * 1000)
+            except (ValueError, AttributeError) as e:
+                logger.warning(f"Failed to parse timestamp '{timestamp}': {e}")
+                return None
+
+        return None
+
+    def normalize_timestamps_in_dict(self, data: Dict[str, Any],
+                                     timestamp_fields: List[str] = None) -> Dict[str, Any]:
```

Ruff (0.15.9)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)